ui/pages/setup_secure.tsx

```diff
@@ -80,12 +80,11 @@ const SetupSecure: NextPage = () => {
           },
         })
         .then(({ data }) => {
-          if ("redirect_to" in data) {
-            window.location.href = data.redirect_to as string;
-            return;
-          }
           if (data?.return_to) {
-            window.location.href = data?.return_to;
+            const flowParam = data.return_to.endsWith("/setup_complete")
+              ? ""
+              : `?flow=${data.id}`;
+            window.location.href = `./setup_complete${flowParam}`;
```

Comment:
What if the return_to page is not `setup_complete` but something else? Shouldn't this be:
```suggestion
            window.location.href = `./${data.return_to}${flowParam}`;
```
?

Reply:
you are right